src/core/js/subviews.js

```diff
@@ -0,0 +1,330 @@
+                childList: true,
+                subtree: true
+            });
+        },
+
+        /**
+         * Swap placeholders for subview elements, debounced
+         */
+        updatePlaceholders: function() {
+            var $placeholders = $('body').find("[data-adapt-subview]");
+            if (!$placeholders.length) return;
+            $placeholders.each(function(index, placeholder) {
+                var $placeholder = $(placeholder);
+                var uid = parseInt($placeholder.attr("data-adapt-subview"));
+                var invocation = this.findByUID(uid);
+                if (!invocation || !invocation.hasInstance()) return;
+                // Swap subview placeholder for subview element
+                invocation.instance.replaceElement($placeholder);
+            }.bind(this));
+        },
+
+        /**
+         * Create an invocation item from the handlebars defined context
+         * @param  {Object} context Handlebars context, containing and model id and subview class name
+         * @return {Invocation}     Newly created or existing invocation
+         */
+        create: function(context) {
+            var newInvocation = new Invocation(context);
+            var existingInvocations = this.items.filter(function(entry) {
+                if (!entry.isEqual(newInvocation)) return;
+                return true;
+            });
+            if (!existingInvocations.length) {
+                // If no existing and matching invocations are available, then
+                // use the newly created invocation and create a subview instance
+                this.items.push(newInvocation);
+                this.updatePlaceholders();
+                return newInvocation.createInstance();
+            }
+            if (existingInvocations.length > 1) {
+                // If more than one invocation matched, then warn that there could
+                // be rendering issues
+                throw "SubViews: Overlapping views for name '"+newInvocation.name+"'";
+            }
+            // Reuse the single pre-existing invocation and its subview instance
+            this.updatePlaceholders();
+            return existingInvocations[0];
+        },
+
+        /**
+         * Get an invocation from the current list based upon its unique id
+         * @param  {Number} uid Placeholder defined invocation uid
+         * @return {Invocation} Found invocation object
+         */
+        findByUID: function(uid) {
+            return _.find(this.items, function(entry) {
+                if (entry.uid !== uid) return;
+                return entry;
+            });
+        },
+
+        /**
+         * Destroy all invocations and associated view instances
+         */
+        destroy: function() {
+            // List length is liable to change during remove procedure
+            // perform in reverse order.
+            for (var i = this.items.length-1; i > -1; i--) {
```

Comment:
The list must be performed in reverse order as the items are removed from the list as it progresses.

Reply:
> I'm not sure how forEach handles a changing length

Good point, I assumed it would be able to handle it. If we're unsure let's leave as-is, it's no big thing.